src/lib/components/PreviewButton.js

```diff
@@ -0,0 +1,92 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2021 CERN.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Icon } from 'semantic-ui-react';
+import { ActionButton } from 'react-invenio-forms';
+
+import { submitAction } from '../state/actions';
+import {
+  FORM_SAVE_FAILED,
+  FORM_SAVE_PARTIALLY_SUCCEEDED,
+  FORM_SAVE_SUCCEEDED,
+  FORM_SAVING,
+} from '../state/types';
+
+export class PreviewButtonComponent extends Component {
+  state = {
+    isLoading: false,
+    previewButtonClicked: false,
+    previousFormState: '',
+  };
+
+  render() {
+    const { record, saveClick, formState, ...uiProps } = this.props;
+    const { isLoading, previewButtonClicked, previousFormState } = this.state;
+    if (previewButtonClicked && formState !== previousFormState) {
+      if (formState === FORM_SAVING) {
+        this.setState({
+          previousFormState: formState,
+        });
+      }
+      if (formState === FORM_SAVE_SUCCEEDED) {
+        window.location = `/records/${record.id}?preview=1`;
+      } else if (
```

Comment:
minor: Let's remove the `else` here or add it on line 36... or better yet use a switch statement:

```js
switch (formState) {
  case FORM_SAVING:
    this.setState({previousFormState: formState});
    break;
  case FORM_SAVE_SUCCEEDED:
    window.location = `/records/${record.id}?preview=1`;
    break;
  case FORM_SAVE_PARTIALLY_SUCCEEDED:
  case FORM_SAVE_FAILED:
    this.setState({
          isLoading: false,
          previewButtonClicked: false,
          previousFormState: formState,
        });
    break;
}
```